cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -240,11 +240,20 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 		saTokenControllerInitFunc := serviceAccountTokenControllerStarter{rootClientBuilder: rootClientBuilder}.startServiceAccountTokenController
 
 		// start client to resource providers
-		if len(c.ResourceProviderClients) > 0 {
+		userAgent := getUserAgent(controllerContext)
```

Comment:
i can see a lot changes around the KCM config struct change is to set the RP and TP KCM with different agent name. probably it is not needed to differentiate them anyway if the user agent is set at the node and ipam controller ( i.e. node related ) already.

Reply:
> i can see a lot changes around the KCM config struct change is to set the RP and TP KCM with different agent name. probably it is not needed to differentiate them anyway if the user agent is set at the node and ipam controller ( i.e. node related ) already.

The original way to create additional restclients does not allow user agent to be set.